src/main/java/org/aszjch/demoapp/domain/article/ArticleServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.aszjch.demoapp.domain.article;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.article.port.ArticlePublisher;
+import org.aszjch.demoapp.domain.article.port.ArticleRepository;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileAssignmentService;
+
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ArticleServiceImpl implements ArticleService, ArticleFileAssignmentService {
+
+    private final ArticleRepository repository;
+    private final ArticlePublisher publisher;
+
+    @Override
+    public List<Article> get() {
+        return repository.findAll();
+    }
+
+    @Override
+    public Article create(final Article article) {
+        final Article saved = repository.save(article);
+        publisher.publish(new ArticleCreatedDto(saved.getId(), saved.getTitle(), saved.getAuthor(),
+                                                saved.getAbstractText()));
+        return saved;
+    }
+
+    @Override
+    public Optional<Article> getById(final Long id) {
+        return repository.findById(id);
+    }
+
+    @Override
+    public Article update(final Long id, final Article article) {
+        final Article entity = getById(id).orElse(new Article());
+        entity.setId(id);
```

Comment:
setId jest tutaj okej, ale ta reszta wygląda jak jakieś mapowanie trochę - wrzuć do nowej metody klasy Article z parametrem Article :) przypisywanie wszystkiego poniżej. Plus findById() powinno zwracać optionala i powinniśmy być w stanie ładnie to w Optionalu obsłużyć wszystko :)

Reply:
Możliwe że będzie można użyć mapstructa żeby nadpisał tylko istniejące w obiekcie wejściowym pola - wtedy połączymy oba obiekty ładnie :)